db/migrate/20171018122656_add_edm_is_shown_by_to_gallery_images.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class AddEdmIsShownByToGalleryImages < ActiveRecord::Migration
+  class GalleryImage < ActiveRecord::Base
+    validates :gallery_id, presence: true
+    validates :europeana_record_id,
+              presence: true, format: { with: Europeana::Record::ID_PATTERN }
+    validates :edm_is_shown_by, presence: true
+
+    def set_edm_is_shown_by
+      api_query = Europeana::Record.search_api_query_for_record_ids([europeana_record_id])
+      record = Europeana::API.record.search(query: api_query, profile: 'rich', rows: 1)['items'].first
+      self.edm_is_shown_by = record['edmIsShownBy'].first
+      save
+    rescue
+      delete
+    end
+  end
+
+  def up
+    add_column :gallery_images, :edm_is_shown_by, :text
+
+    GalleryImage.all.each(&:set_edm_is_shown_by)
```

Comment:
The performance of this migration can be improved by a factor of 100 by querying the API for 100 records at a time:
```ruby
GalleryImage.find_in_batches(batch_size: 100) do |batch|
  record_ids = batch.map(&:europeana_record_id)
  api_query = Europeana::Record.search_api_query_for_record_ids(record_ids)
  response = Europeana::API.record.search(query: api_query, profile: 'rich', rows: 100)

  batch.each do |image|  
    item = response['items'].detect { |item| item['id'] == image.europeana_record_id }
    if item.nil?
      image.delete
    else
      image.edm_is_shown_by = item['edmIsShownBy'].first
      image.save
    end
  end
end
```